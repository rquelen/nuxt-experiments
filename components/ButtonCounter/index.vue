<template>
  <button @click="incrementCounter" class="test">Button data: {{ counter }}</button>
</template>

<script>
export default {
  name: 'ButtonCounter',
  data () {
    return {
      counter: 0,
    };
  },
  methods: {
    incrementCounter: function () {
      this.counter += 1;
      this.$emit('increment');
    },
  },
};
</script>

<style scoped>
  .test {
    font-size: 30px;
  }
</style>
