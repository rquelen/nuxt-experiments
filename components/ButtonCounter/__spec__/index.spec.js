import {shallow} from '@vue/test-utils';
import ButtonCounter from '../index';

describe('ButtonCounter', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallow(ButtonCounter);

    // then
    expect(wrapper.element).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should initialize counter to 0', () => {
    // when
    const wrapper = shallow(ButtonCounter);

    // then
    const button = wrapper.find('.test');
    expect(button.text()).toContain('0');
  });

  it('should increment counter on click', () => {
    // given
    const wrapper = shallow(ButtonCounter);
    const button = wrapper.find('.test');

    // when
    button.trigger('click');

    // then
    expect(button.text()).toContain('1');
  });

  it('should emit an event increment on click', () => {
    // given
    const wrapper = shallow(ButtonCounter);
    const button = wrapper.find('.test');
    const spy = jest.fn();
    wrapper.vm.$on('increment', spy);

    // when
    button.trigger('click');

    // then
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
