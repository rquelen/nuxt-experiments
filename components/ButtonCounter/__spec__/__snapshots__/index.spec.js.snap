// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCounter should match snapshot 1`] = `
<button
  class="test"
>
  Button data: 0
</button>
`;
